src/core/Akka.Tests/Actor/AskSpec.cs

```diff
@@ -66,52 +66,63 @@ protected override void OnReceive(object message)
+            var res = await actor.Ask<string>("answer", TimeSpan.FromSeconds(10));
+            res.ShouldBe("answer");
         }
 
         [Fact]
-        public void Can_get_timeout_when_asking_actor()
+        public async Task Can_get_timeout_when_asking_actor()
         {
             var actor = Sys.ActorOf<SomeActor>();
-            Assert.Throws<AggregateException>(() => { actor.Ask<string>("timeout", TimeSpan.FromSeconds(3)).Wait(); });
+            await Assert.ThrowsAsync<AskTimeoutException>(async () => await actor.Ask<string>("timeout", TimeSpan.FromSeconds(3)));
         }
 
         [Fact]
-        public void Can_cancel_when_asking_actor()
-        {            
+        public async Task Can_cancel_when_asking_actor()
+        {
             var actor = Sys.ActorOf<SomeActor>();
-            var cts = new CancellationTokenSource(TimeSpan.FromSeconds(3));
-            Assert.Throws<AggregateException>(() => { actor.Ask<string>("timeout", Timeout.InfiniteTimeSpan, cts.Token).Wait(); });
-            Assert.True(cts.IsCancellationRequested);
+            using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(3)))
+            {
+                await Assert.ThrowsAsync<TaskCanceledException>(async () => await actor.Ask<string>("timeout", Timeout.InfiniteTimeSpan, cts.Token));
+            }
         }
+
         [Fact]
-        public void Cancelled_ask_with_null_timeout_should_remove_temp_actor()
+        public async Task Cancelled_ask_with_null_timeout_should_remove_temp_actor()
         {
             var actor = Sys.ActorOf<SomeActor>();
-            var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(100));
-            Assert.Throws<AggregateException>(() => { actor.Ask<string>("cancel", cts.Token).Wait(); });
-            Assert.True(cts.IsCancellationRequested);
+
+            using (var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(100)))
+            {
+                await Assert.ThrowsAsync<TaskCanceledException>(async () => await actor.Ask<string>("cancel", cts.Token));
+            }
+
             Are_Temp_Actors_Removed(actor);
         }
+
         [Fact]
-        public void Cancelled_ask_with_timeout_should_remove_temp_actor()
+        public async Task Cancelled_ask_with_timeout_should_remove_temp_actor()
         {
             var actor = Sys.ActorOf<SomeActor>();
-            var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(100));
-            Assert.Throws<AggregateException>(() => { actor.Ask<string>("cancel", TimeSpan.FromSeconds(30), cts.Token).Wait(); });
-            Assert.True(cts.IsCancellationRequested);
+            using (var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(100)))
+            {
+                await Assert.ThrowsAsync<TaskCanceledException>(async () => await actor.Ask<string>("cancel", TimeSpan.FromSeconds(30), cts.Token));
+            }
+
             Are_Temp_Actors_Removed(actor);
```

Comment:
@Aaronontheweb @Horusiath What is this thing doing here? 
We kinda testing Ask here. As far as I can see it should not affect an actor it's used upon.

Reply:
Internally in order to work `Ask` makes a lightweight temporary actor reference. One of the bugs in the past was that those temporary actors where not recycled after `Ask` task completed, causing memory leaks.